custom/src/main/java/com/splunk/opentelemetry/servicename/JettyServiceNameDetector.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.servicename;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class JettyServiceNameDetector extends AppServerServiceNameDetector {
+  private static final Logger log = LoggerFactory.getLogger(JettyServiceNameDetector.class);
+
+  JettyServiceNameDetector(ResourceLocator locator) {
+    super(locator, "org.eclipse.jetty.start.Main", false);
+  }
+
+  @Override
+  boolean isValidAppName(Path path) {
+    // jetty deployer ignores directories ending with ".d"
+    if (Files.isDirectory(path)) {
+      return !path.getFileName().toString().endsWith(".d");
+    }
+    return true;
+  }
+
+  @VisibleForTesting
+  static Path parseJettyBase(String programArguments) {
+    if (programArguments == null) {
+      return null;
+    }
+    int start = programArguments.indexOf("jetty.base=");
+    if (start == -1) {
+      return null;
+    }
+    start += "jetty.base=".length();
+    if (start == programArguments.length()) {
+      return null;
+    }
+    // Take the path until the first space. If the path doesn't exist extend it up to the next
+    // space. Repeat until a path that exists is found or input runs out.
+    int next = start;
+    while (true) {
+      int nextSpace = programArguments.indexOf(' ', next);
+      if (nextSpace == -1) {
+        Path candidate = Paths.get(programArguments.substring(start));
+        return Files.exists(candidate) ? candidate : null;
+      }
+      Path candidate = Paths.get(programArguments.substring(start, nextSpace));
+      next = nextSpace + 1;
+      if (Files.exists(candidate)) {
+        return candidate;
+      }
+    }
```

Comment:
I totally think that this is fine for now, but I suspect there's some remotely possible edge case out there that could cause this to parse the commandline string differently than a shell might have parsed it, resulting in a path that includes more than the user specified. I'm speculating, but imagine a dir with two subdirs

```
jetty
   +--- jetty base 
   +--- jetty base dev
```

If the user runs a commandline like
```
jetty <whatever> "jetty.base=jetty base dev" jetty.foo=bar
```
and if the JVM reconstitutes this into the `sun.java.command` as 
```
jetty <whatever> jetty.base=jetty base dev jetty.foo=bar
``` `
then I _think_ this will end up using `jetty base` as the path even tho the user intended `jetty base dev`. 

Seems very unlikely that this will happen, so I dunno if it's worth trying to fix now. #famouslastwords

Reply:
It should be possible to do it accurately with https://docs.oracle.com/javase/9/docs/api/java/lang/ProcessHandle.Info.html#arguments-- on jdk9

Reply:
Yeah totally. A change for another time tho.